src/loose/statement.js

```diff
@@ -218,9 +220,12 @@ lp.parseFor = function(node, init) {
   return this.finishNode(node, "ForStatement")
 }
 
-lp.parseForIn = function(node, init) {
+lp.parseForIn = function(node, init, isAwait) {
```

Comment:
I think I like Adrian's [simpler solution](https://github.com/acornjs/acorn/pull/661/files#diff-a79fa4b688420e7448146c4c95c28ca6R54) (where the caller directly sets this flag in the node) better.